helix-term/src/ui/editor.rs

```diff
@@ -427,7 +427,7 @@ impl EditorView {
 
         let current_line = doc
             .text()
-            .char_to_line(doc.selection(view.id).primary().anchor);
+            .char_to_line(doc.selection(view.id).primary().head);
```

Comment:
You actually want to use the `.cursor()` method here to get the position of the block-cursor, not `.head`.  `.head` returns the position of the *edge* of the range, which isn't quite the same as the cursor.